apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -291,10 +292,7 @@ export class CaseController {
     const hasCourtIdUpdate =
       theCase.courtId && update.courtId && theCase.courtId !== update.courtId
     if (hasCourtIdUpdate) {
-      update.courtCaseNumber = null
-      update.judgeId = null
-      update.registrarId = null
-      update.state = CaseState.SUBMITTED
+      transitionCase(CaseTransition.SUBMIT, theCase, user)
```

Comment:
I think it makes more sense to call this transition here when a certain change occurs in a case update compared to calling the transition case endpoint in from the client. In the client the option to change the court occurs in at least two places. 

Still making this call here or in the client, it will trigger 2x case updates compared to the previous update assignment.